processing/src/main/java/org/apache/druid/query/aggregation/SerializablePairLongFloatSerde.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation;
+
+import org.apache.druid.common.config.NullHandling;
+
+import java.nio.ByteBuffer;
+
+/**
+ * The class serializes a Long-Float pair (SerializablePair<Long, Float>).
+ * The serialization structure is: Long:Float
+ * <p>
+ * The class is used on first/last Float aggregators to store the time and the first/last Float.
+ * Long:Float -> Timestamp:Float
+ */
+public class SerializablePairLongFloatSerde extends AbstractSerializablePairSerde<SerializablePairLongFloat>
+{
+  public static final String TYPE_NAME = "serializablePairLongFloat";
+
+  public SerializablePairLongFloatSerde()
+  {
+    super(SerializablePairLongFloat.class);
+  }
+
+  @Override
+  public String getTypeName()
+  {
+    return TYPE_NAME;
+  }
+
+  @Override
+  protected SerializablePairLongFloat toPairObject(ByteBuffer buffer)
+  {
+    final ByteBuffer readOnlyBuffer = buffer.asReadOnlyBuffer();
+    long lhs = readOnlyBuffer.getLong();
+    boolean isNotNull = readOnlyBuffer.get() == NullHandling.IS_NOT_NULL_BYTE;
+    if (isNotNull) {
+      return new SerializablePairLongFloat(lhs, readOnlyBuffer.getFloat());
+    } else {
+      return new SerializablePairLongFloat(lhs, null);
+    }
+  }
+
+  @Override
+  protected byte[] pairToBytes(SerializablePairLongFloat val)
```

Comment:
Since the super class says `val` can be null, all these implementations should be able to handle a null `val`.

I haven't dug in yet to know what this means, but this same pattern exists in all 3 implementations.

Reply:
in the new commit, null check is added to the caller of this method, so no need to handle null in this method